.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,84 @@
+name: Build Docker image
+
+on: [push]
+
+env:
+  PARABOL_DOCKERFILE: docker/parabol-ubi/docker-build/dockerfiles/pipeline.dockerfile
+  PARABOL_BUILD_ENV_PATH: docker/parabol-ubi/docker-build/environments/pipeline
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    services:
+      rethinkdb:
+        image: rethinkdb:2.4.0
+        ports:
+          - 8080:8080
+          - 28015:28015
+          - 29015:29015
+      redis:
+        image: redis:6.2.6
+        ports:
+          - 6379:6379
+      postgres:
+        image: postgres:12.10-alpine
+        env:
```

Comment:
+1 could we do a `env_file: ./environments/pipeline` to keep a single source of truth?

Reply:
Sadly we cannot do that. Please check the official documentation of Github Workflows for [job services](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idservices). The only reference to environment variables is just [env](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idservicesservice_idenv). And we cannot pass the env vars to the [PostgreSQL image](https://hub.docker.com/_/postgres) mounting a file somewhere.

Reply:
could we use a github action to do that job?
e.g.
```yml
- uses: cardinalby/export-env-action@v2
  with:
    envFile: 'constants.env'  
```

another option (untetsed) is https://github.com/marketplace/actions/dotenv-action

Reply:
We could, but that action (untested on my side as well), uses an env file to set environment variables on the Github runner host. The services are executed BEFORE any action. They run on the host, but before you can run anything else.

What is the tension here? This variables are unlikely to be changed as they are just temporary credentials for build. I don't see why would we want to change or adapt them. At least not very often.

Reply:
if the env vars live on the host that seems OK with me!

The main concern is readability & knowledge transfer. There is an implicit constraint that the password used here must equal the password used in the env file. You know that & I know that, **_but the new developer we hire won't know that._** That constraint isn't documented anywhere. We could add a comment saying the two must be equal, but a better solution is self-documenting code. If this file reads the variables from the env, it communicates that dependency without extra words.

Reply:
100% with you on that. I tried solving this using a pipeline.env. 

I think I didn't explain myself correctly about the action you proposed. It wouldn't work. It sets the vars on the runner AFTER the services are running. It is either not using services, which are the easiest way of running the dbs in the pipelines, or having this vars hardcoded (and add a comment, which looks like a fine solution for a simple case like this one to me)

Reply:
oh! gotcha, `uses` isn't available under `services`? maaaaan that's lame. ah well, a comment works!